packages/opentelemetry-metrics/src/Meter.ts

```diff
@@ -91,6 +95,32 @@ export class Meter implements types.Meter {
       ...DEFAULT_METRIC_OPTIONS,
       ...options,
     };
-    return new GaugeMetric(name, opt);
+    const gauge = new GaugeMetric(name, opt);
+    this._registerMetric(name, gauge);
+    return gauge;
+  }
+
+  /**
+   * Gets a collection of Metric`s to be exported.
+   * @returns The list of metrics.
+   */
+  getMetrics(): ReadableMetric[] {
+    return Array.from(this._metrics.values())
+      .map(metric => metric.get())
+      .filter(metric => !!metric) as ReadableMetric[];
+  }
+
+  /**
+   * Registers metric to register.
+   * @param name The name of the metric.
+   * @param metric The metric to register.
+   */
+  private _registerMetric<T>(name: string, metric: Metric<T>): void {
+    if (this._metrics.has(name)) {
+      throw new Error(
```

Comment:
I remember a conversation about the tracer never throwing exceptions, which I assume applies to metrics too. Could this be an error log instead?

Reply:
One option I could think is log an error and return `NoopMetric`. But If the `logger` is disabled or using noop implementation, it would be confusing for users as recorded data won't get exported to backend. WDYT?

Reply:
I think we should log an error or log as an info and replaces the old registered metric by the new

Reply:
Yeah, I think either behavior would be preferable to throwing an exception, which could crash the request (assuming we are working with the "tracers should never throw" principle, which I personally think makes sense)

Reply:
I agree with Oliver here. Replacing the old metric with the new one and logging an error seems to be appropriate

Reply:
Replaced `throw new Error` with `logger.error`, but replacing the old registered metric by the new seems to be little problematic, especially when you use the same name but different `type`  or `unit`. For now, I just log an error and not do anything. I will open an issue to discuss it separately.